exonum-java-binding/core/src/main/java/com/exonum/binding/core/runtime/ServiceRuntimeAdapter.java

```diff
@@ -75,29 +75,33 @@ void initialize(long nodeNativeHandle) {
   /**
    * Deploys the Java service artifact.
    *
-   * @param name the Java service artifact name in format "groupId:artifactId:version"
+   * @param artifactId bytes representation of the service artifact
    * @param deploySpec the deploy specification as a serialized
    *     {@link com.exonum.binding.core.runtime.DeployArguments}
    *     protobuf message
    * @throws IllegalArgumentException if the deploy specification or id are not valid
    * @throws ServiceLoadingException if the runtime failed to load the service or it is not correct
    * @see ServiceRuntime#deployArtifact(ServiceArtifactId, String)
    */
-  void deployArtifact(String name, byte[] deploySpec) throws ServiceLoadingException {
-    DeployArguments deployArguments = parseDeployArgs(name, deploySpec);
+  void deployArtifact(byte[] artifactId, byte[] deploySpec) throws ServiceLoadingException {
+    ArtifactId artifact = parseArtifact(artifactId);
```

Comment:
do we need to check  here that artifact.runtimeId is java runtime?

Reply:
I believe it is surely Core's responsibility

Reply:
I'd say the core responsibility is to _pass_ Java artifact ids only 🙃 But the check is redundant — Java Runtime loads only Java artifacts; if you pass non-Java artifact id later, it just won't be found.